ENIGMAsystem/SHELL/Universal_System/object.cpp

```diff
@@ -77,22 +77,22 @@ namespace enigma
     bool object_basic::can_cast(int obj) const { return false; }
 
     extern objectstruct objs[];
-    extern int obj_idmax;
+    extern size_t object_idmax;
 
     void objectdata_load()
     {
-        objectdata = new objectstruct*[obj_idmax];
+        objectdata = new objectstruct*[object_idmax];
         for (int i = 0; i < objectcount; i++)
             objectdata[objs[i].id] = &objs[i];
     }
 }
 
 #if defined(SHOW_ERRORS) && SHOW_ERRORS
   #define errcheck(objid,err) \
-  if (objid < 0 or objid >= enigma::objectcount or !enigma::objectdata[objid]) \
+  if (objid < 0 or size_t(objid) >= enigma::object_idmax or !enigma::objectdata[objid]) \
```

Comment:
This pattern's been bugging me, but I've yet to reach a satisfying conclusion, so I'll mention it: the latter check technically renders the former unnecessary. EXCEPT... the compiler should know that, anyway, and remove it, and historically, programmers getting clever about the storage format of their data has *seriously* fucked with optimizations. So, let's leave this as you now have it... but cry a little about the state of this world.

Reply:
I could do my usual routine of hobnobbing you here, but it generally adds little value, so just pretend I have done that. Maybe we can cry about it less after #1125 is finally done.